### PR TITLE
Collect device data concurrently on Windows

### PR DESCRIPTION
**Pull request by codingllama**
Change data collection to run exec-ed commands concurrently.

Changelog: Collect device trust data concurrently on Windows

Closes https://github.com/gravitational/teleport.e/issues/1481.

### CODE REVIEW

File: lib/devicetrust/native/device_windows.go [hunk 1]
```diff
@@ -177,30 +178,41 @@ func getOSBuildNumber() (string, error) {
...
+				return nil // Swallowed on purpose.
+			}
+
+			*spec.out = val
+			return nil
+		})
 	}
+
+	// We want to fetch as much info as possible, so errors are ignored.
+	_ = g.Wait()
```
[the remark this answers is not in the document]
Comment (codingllama) [reply lib/devicetrust/native/device_windows.go:214]: Yes, the backend will decide what's important. A lot of this info is not mandatory (we do need one good serial number, but that's it).

[the remark this answers is not in the document]
Comment (codingllama) [reply lib/devicetrust/native/device_windows.go:214]: You'll get something like these back:

* https://github.com/gravitational/teleport.e/blob/6be8010aef6114e1110c17a93661223372f2e267/lib/devicetrust/devicetrustv1/authn.go#L103
* https://github.com/gravitational/teleport.e/blob/master/lib/devicetrust/devicetrustv1/enroll.go#L73

[the remark this answers is not in the document]
Comment (codingllama) [reply lib/devicetrust/native/device_windows.go:214]: You'll get something like these back:

* https://github.com/gravitational/teleport.e/blob/6be8010aef6114e1110c17a93661223372f2e267/lib/devicetrust/devicetrustv1/authn.go#L103
* https://github.com/gravitational/teleport.e/blob/master/lib/devicetrust/devicetrustv1/enroll.go#L73

(Ie, it says "device serial number required".)

I think it's reasonably clear, but if you are still confused we'll likely ask for the `tsh -d` logs anyway. I'd rather make the client resilient, without trying to be too clever, and let the server decide what matters and what doesn't.

[the remark this answers is not in the document]
Comment (codingllama) [reply lib/devicetrust/native/device_windows.go:214]: WDYT? Still rather have the old behavior?

**Review by rosstimothy — APPROVED** [review]

**Review by strideynet — APPROVED** [review]

**Comment by codingllama** [comment]
Thanks for the quick review, Noah!

**Pull request merged**